microsetta_admin/server.py

```diff
@@ -774,6 +775,217 @@ def scan():
                                       recorded_type)
 
 
+def _get_color_code():
+    val = '0123456789ABCDEF'
+    code = ["#"+''.join([random.choice(val) for i in range(6)])]
+    return code[0]
+
+
+def _get_legends(criteria):
+
+    dict = {}
+    if criteria not in session:
+        status, fields = APIRequest.get(
+            '/api/admin/barcode_query_fields')
+
+        if status == 200:
+            for obj in fields:
+                if criteria == "Type" and obj["label"] == "Sample Site":
```

Comment:
Is it necessary to have both the criteria and obj["label"] clauses in these if statements?

Reply:
Simplified the code further to remove code duplications and eliminate both clauses from if statements.